lyncs_io/tar.py

```diff
@@ -0,0 +1,245 @@
+
+
+def _load(paths, tar, **kwargs):
+    new_path_dict = nested_dict()
+    for path in paths:
+        parts = path.split("/")
+        if parts:
+            marcher = new_path_dict
+            for key in parts[:-1]:
+                marcher = marcher[key]
+            header = _load_member(
+                tar, find_member(tar, tar.getmember(path).name), as_data=True, **kwargs
+            )
+            marcher[parts[-1]] = header
+    return default_to_regular(new_path_dict)
+
+
+def _load_dispatch(tar, key, loader, depth=1, all_data=False, **kwargs):
+    # if .h5
+    if key and key.split("/")[0].endswith(".h5"):
+        raise NotImplementedError(
+            f"HDF5 file {key} is not supported when inside a tarball"
+        )
+
+    if is_dir(tar, format_key(key)):
+        key = format_key(key)
+        paths = [
+            member.name
+            for member in tar.getmembers()
+            if (
+                (member.name.startswith(key) or key == "/")
+                and (get_depth(member.name, key) <= depth or all_data)
+            )
+        ]
+
+        # avoid {dir : {data}}. Return {data} instead if key is given.
+        _dict = (
+            _load(paths, tar, **kwargs)[key[:-1]]
+            if key != "/"
+            else _load(paths, tar, **kwargs)
+        )
+
+        return Archive(_dict, loader=loader, path=key)
+
+    return _load_member(tar, find_member(tar, key), **kwargs)
+
+
+def load(filename, key=None, chunks=None, comm=None, **kwargs):
+    """
+    Load function for tar
+    """
+
+    if chunks and comm:
+        raise ValueError("chunks and comm parameters cannot be both set")
+
+    filename, key = split_filename(filename, key)
+    mode_suffix = _get_mode(filename)
+    kwargs = {"comm": comm, **kwargs}
+    loader = Loader(load, filename, kwargs=kwargs)
+
+    with tarfile.open(filename, "r" + mode_suffix) as tar:
+        return _load_dispatch(tar, key, loader, **kwargs)
+
+
+def head(*args, **kwargs):
+    """
+    Head function for tar
+    """
+    return load(*args, header_only=True, **kwargs)
+
+
+def _get_mode(filename):
+    """
+    Returns the mode (from modes) with which the tarball should be read/written as
+    """
+    ext = splitext(filename)[-1]
+    for key, val in modes.items():
+        if ext in val:
+            return key
+    raise ValueError(f"{ext} is not supported.")
+
+
+def is_dir(tar, key):
+    """
+    Check whether a member in a tarball is a directory
+    """
+    for member in tar.getmembers():
+        if key == "/" or member.name.startswith(key):
+            return True
+    return False
+
+
+@contextmanager
+def _extract(tar, member, get_buff=False, **kwargs):
+    if kwargs.get("comm"):
+        with tempdir_MPI() as temp:
+            check_comm(kwargs["comm"])
+            tar.extract(member, path=temp)
+            yield temp + "/" + listdir(temp)[0]
+    elif get_buff:
+        raise NotImplementedError
```

Comment:
The purpose of get_buff is to `yield tar.extractfile(member)` directly, right?
So let's have it implemented but not used for now.
Then we should do a list of formats that can use directly `tar.extractfile(member)` without reading it.